conan_docker_tools/application.py

```diff
@@ -0,0 +1,71 @@
+"""Execute all stages
+"""
+from variables import Variables
+from compiler import GccCompiler, ClangCompiler
+from image import Image
+from build import Build
+from test import Test
+from deploy import Deploy
+from arguments import Arguments
+import logging
+
+
+class Application(object):
+
+    def __init__(self):
+        logging.basicConfig(format='%(levelname)s: %(message)s', level=logging.DEBUG)
+        logging.info("Starting Conan Docker Tools")
+        self.__arguments = Arguments()
+        self.__variables = Variables()
+
+    def run(self):
+        gcc_versions, clang_versions = self.__get_compiler_versions()
+        self.__process_compiler(GccCompiler(), gcc_versions)
+        self.__process_compiler(ClangCompiler(), clang_versions)
+
+    def __process_compiler(self, compiler, versions):
```

Comment:
For private methods we use only one "_" underscore by convention, it allows the IDE etc to access the code.